scripts/pre_commit_linter.py

```diff
@@ -630,1300 +875,1151 @@ def _lint_py_files(config_pylint, config_pycodestyle, files_to_lint, result):
+                summary_messages.append(summary_message)
+            else:
+                summary_message = (
+                    '%s   Docstring check passed' % _MESSAGE_TYPE_SUCCESS)
+                print summary_message
+                summary_messages.append(summary_message)
 
+        return summary_messages
 
-def _check_html_directive_name(all_files):
-    """This function checks that all HTML directives end
-    with _directive.html.
-    """
-    print 'Starting HTML directive name check'
-    print '----------------------------------------'
-    total_files_checked = 0
-    total_error_count = 0
-    files_to_check = [
-        filename for filename in all_files if not
-        any(fnmatch.fnmatch(filename, pattern) for pattern in EXCLUDED_PATHS)
-        and filename.endswith('.js')]
-    failed = False
-    summary_messages = []
-    # For RegExp explanation, please see https://regex101.com/r/gU7oT6/37.
-    pattern_to_match = (
-        r'templateUrl: UrlInterpolationService\.[A-z\(]+' +
-        r'(?P<directive_name>[^\)]+)')
-    with _redirect_stdout(_TARGET_STDOUT):
-        for filename in files_to_check:
-            content = FileCache.read(filename)
-            total_files_checked += 1
-            matched_patterns = re.findall(pattern_to_match, content)
-            for matched_pattern in matched_patterns:
-                matched_pattern = matched_pattern.split()
-                directive_filename = ''.join(matched_pattern).replace(
-                    '\'', '').replace('+', '')
-                if not directive_filename.endswith('_directive.html'):
-                    failed = True
-                    total_error_count += 1
-                    print (
-                        '%s --> Please ensure that this file ends'
-                        'with _directive.html.' % directive_filename)
-                    print ''
+    def _check_comments(self):
+        """This function ensures that comments follow correct style."""
+        print 'Starting comment checks'
+        print '----------------------------------------'
+        summary_messages = []
+        files_to_check = [
+            filename for filename in self.all_files if not
+            any(fnmatch.fnmatch(filename, pattern) for pattern in
+                EXCLUDED_PATHS) and filename.endswith('.py')]
+        message = 'There should be a period at the end of the comment.'
+        failed = False
+        space_regex = re.compile(r'^#[^\s].*$')
+        capital_regex = re.compile('^# [a-z][A-Za-z]* .*$')
+        with _redirect_stdout(_TARGET_STDOUT):
+            for filename in files_to_check:
+                file_content = FileCache.readlines(filename)
+                file_length = len(file_content)
+                for line_num in range(file_length):
+                    line = file_content[line_num].strip()
+                    next_line = ''
+                    previous_line = ''
+                    if line_num
```

Comment:
Can you add this part to the function `_print_complete_summary_of_errors` itself and remove the usage of the variable `error_messages`?

Reply:
Done.